myuw/static/js/banner/profile.js

```diff
@@ -2,6 +2,33 @@ var Profile = {
     render_upon_data: function() {
         var source = $("#profile-content").html();
         var template = Handlebars.compile(source);
+
+        // Process data to determine if there are any pending majors or minors
+        var profile_data = WSData.profile_data();
+
+        if(profile_data.term_majors && profile_data.term_majors.length > 0){
```

Comment:
can you track these 2 things in the python side, and just set them at the top level of your data structure?  i think you can use 2-4 lines or so of python to drop 26 lines here